core/src/main/java/org/jivesoftware/sparkimpl/certificates/CertificateController.java

```diff
@@ -100,14 +98,57 @@ public CertificateController(LocalPreferences localPreferences) {
 					certEntry[0] = cert.getSubject();
 				}
 				certEntry[1] = cert.getValidityStatus();
-				certEntry[2] = cert.isExempted();
+				certEntry[2] = isOnExceptionList(cert.getAlias());
 				tableModel.addRow(certEntry);
 			}
 		}
 	}
 	
-	public void isOnExceptionList(X509Certificate cert){
+	
+	/**
+	 * If argument is true then move certificate to the exceptions Keystore, if false then move to the trusted Keystore.
+	 * Useful for checkboxes where it's selected value indicates where certificate should be moved.
+	 * @param checked
+	 */
+	public void addOrRemoveFromExceptionList(boolean checked){
+		if (checked) {
+			try {
+				moveCertificate(TRUSTED, EXCEPTIONS);
+			} catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException ex) {
+				Log.error("Error at moving certificate from trusted list to the exception list", ex);
+			}
+		} else {
+			try {
+				moveCertificate(EXCEPTIONS, TRUSTED);
+			} catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException ex) {
+				Log.error("Error at moving certificate from exceptions list to trusted list", ex);
+			}
+		}
+	}
+	
+	/**
+	 * Return information if certificate is on exception list.
+	 * @param alias
+	 * @return
+	 */
+	public boolean isOnExceptionList(String alias){
+		try (FileInputStream input = new FileInputStream(EXCEPTIONS)) {
+			trustStore = KeyStore.getInstance("JKS");
```

Comment:
This loads the entire keystore file, each time. That's probably very slow. You might want to keep it in memory.